x-pack/plugins/cloud_security_posture/server/routes/benchmark_rules/bulk_action/utils.ts

```diff
@@ -5,17 +5,99 @@
  * 2.0.
  */
 
+import { transformError } from '@kbn/securitysolution-es-utils';
 import type {
   SavedObjectsClientContract,
   SavedObjectsUpdateResponse,
 } from '@kbn/core-saved-objects-api-server';
-import { CspBenchmarkRulesStates, CspSettings } from '../../../../common/types/rules/v3';
+import type { Logger } from '@kbn/core/server';
+import type { FindResult, RulesClient } from '@kbn/alerting-plugin/server';
+import type { RuleParams } from '@kbn/alerting-plugin/server/application/rule/types';
+import type {
+  CspBenchmarkRule,
+  CspBenchmarkRulesStates,
+  CspSettings,
+} from '../../../../common/types/rules/v3';
+import {
+  convertRuleTagsToKQL,
+  generateBenchmarkRuleTags,
+} from '../../../../common/utils/detection_rules';
 
 import {
+  CSP_BENCHMARK_RULE_SAVED_OBJECT_TYPE,
   INTERNAL_CSP_SETTINGS_SAVED_OBJECT_ID,
   INTERNAL_CSP_SETTINGS_SAVED_OBJECT_TYPE,
 } from '../../../../common/constants';
 
+const disableDetectionRules = async (
+  detectionRulesClient: RulesClient,
+  detectionRules: Array<FindResult<RuleParams>>
+) => {
+  const idsToDisable = detectionRules
+    .map((detectionRule) => {
+      return detectionRule.data.map((data) => data.id);
+    })
+    .flat();
+  if (!idsToDisable.length) return;
+  return await detectionRulesClient.bulkDisableRules({ ids: idsToDisable });
+};
```

Comment:
I would like to suggest a small change to isolate our logic from the `detectionRulesClient`
If we will just create a function that returns `idsToDisable`, we can unit test it individually and just call to `await detectionRulesClient.bulkDisableRules({ ids: idsToDisable })` when needed.

example:
```
// export and unit test
export const getRuleIdsToDisable = async (
  detectionRules: Array<FindResult<RuleParams>>
) => {
  const idsToDisable = detectionRules
    .map((detectionRule) => {
      return detectionRule.data.map((data) => data.id);
    })
    .flat();

  return idsToDisable;
};

const ruleIdsToDisable = getRuleIdsToDisable(x)

await detectionRulesClient.bulkDisableRules({ ids: ruleIdsToDisable });
```

note that if `detectionRulesClient.bulkDisableRules({ ids: ruleIdsToDisable })` throws errors when `ruleIdsToDisable` is undefined, we will wrap it with an `if (!!ruleIdsToDisable.length)`

The alternative to that is to add FTRs, but they will be more complicated since we will need to mock/check the detection rules page. I think we should assume they already have their tests on their disable rules api and we should only test our side of the logic.